bots/psuteam7bot/prophet.js

```diff
@@ -37,183 +55,171 @@ prophet.doAction = (self) => {
-            self.potentialEnemyCastleLocation.shift();
-            self.target = self.potentialEnemyCastleLocation[0];
+            if(movement.aStarPathfinding(self, self.me, self.target, false)) 
+            {
+                self.log(self.path);
+            } 
+            else 
+            {
+                self.log('Cannot get path to enemy base');
+                return;
+            }
         }
+        self.log('ATTACKER prophet ' + self.id + ' moving towards enemy base, Current: [' + self.me.x + ',' + self.me.y + ']')
+        return movement.moveAlongPath(self);
+    }
+    //Should not fall through unless attacker with no nearby teammates/ squad
+    self.log('prophet ' + self.role + ' ' + self.me.id + ' doing nothing')
+    return;
+}
 
-        //No more patrol waypoint, 
 
-        if(self.potentialEnemyCastleLocation.length === 0)
-        {
-            //TODO replace with smarter strat
-            //Random movement cost a lot of fuel
-            //Should be in the quadrant opposite to base which is surely enemy quadrant, pick a random direction and move
-            /*
-            const choices = [[0,-1], [1, -1], [1, 0], [1, 1], [0, 1], [-1, 1], [-1, 0], [-1, -1]];
-            let patrolQuadrant = movement.checkQuadrant(self.me, self.map);
-            let destQuadrant = -1;
-            let choice = null;
-
-
-            choice = choices[Math.floor(Math.random()*choices.length)];
-            destQuadrant = movement.checkQuadrant({x: self.me.x+choice.x, y: self.me.y+choice.y}, self.map);
-
-            //Choose opposite direction if trying to go outside patrol quadrant
-            if(destQuadrant !== patrolQuadrant)
-                choice = [choice[0]*-1, choice[1]*-1];
-            
-            self.log("Patrolling enemy quadrant " + patrolQuadrant);
-            return self.move(...choice);
-            */
-           return;
-        }
+prophet.fleeBehavior = (self, visibleRobots) => {
+    let minUnattackable = combat.filterByTeam(self, visibleRobots, -1);
+    minUnattackable = combat.filterByRange(minUnattackable, self.me, 0, SPECS.UNITS[self.me.unit].ATTACK_RADIUS[0]-1);
+
+    
+    //Set 'fleeing behavior?' (whenattack target < min attack range)
+    if(minUnattackable.length > 0)
+    {
+        const attacker = minUnattackable[0];
+        const direction = movement.getRelativeDirection(self.me, attacker);
+        const directionIndex = movement.getDirectionIndex(direction);
+        const moveablePositions = movement.getMoveablePositions(self);
 
-        //Move towards a patrol waypoint as a squad
-        let squad = combat.filterByTeam(self, visibleRobots, self.me.team);
-        squad = combat.filterByRange(squad, self.me, 0, 64);
+        //TODO check for other potential tile at dirIndex+1, dirIndex-1
+        moveablePositions.filter((location) => {
+            if(directionIndex === location.dirIndex && location.r2 === SPECS.UNITS[self.me.unit].SPEED)
```

Comment:
Might not want `location.r2 === SPECS.UNITS[self.me.unit].SPEED` as this will get ONLY things in the right direction and EXACTLY max R2 speed. Practically most moves are less than R2 (i.e. 2 or 1 as opposed to 4) so this will probably return nothing for prophets. Need to fix this

Reply:
I think we may want to filter only move that are perpandicular or opposite the direction of the enemy, but then sort them all by distance and potentially those in the exact opposite direction. This is because the "optimal" fleeing move might be onto an impassible square, so we want to have backup options. Check out my `sort` function in `processAStarCell` for a sorting example